mantidproject/mantid
# Engg GUI: add focusing tab

Fixes #13375. This is a first and incomplete version. More features will be added (see #13498,  #13375)

**To test**:
- this comes with a few additional test cases, they should pass
- code review
- you can do a small manual test (see instructions below)

To use the focus tab: first you need to calibrate, following the steps
explained in #13370 (you need a few files too). If you've done it
before the GUI will remember the last settings.

In the settings tab there's now a 'calibration output folder' that you need to set to 
somewhere when you can write files (if you leave it empty files will be written to your Mantid
executable current directory).

Then go to the focus tab, and give:
- run number: you can try 228061 (the file ENGINX00228061.nxs is available from the unit test data)
  and/or 213855 (the file ENGINX00213855.nxs is available from the doc test data), or other ENGINX runs from the archive.
- Bank: select 1 or 2
- You should see two workspaces in the workspaces window: `engggui_focusing_input_ws`, `engggui_focusing_output_ws` -
  the output should be "focused", that is it should have a single spectrum.

The Engg GUI unit test can be run locally with a command like this `ctest -V -R EnggDiffraction`
